File: solutions/go/series/1/series.go
package series

func All(n int, s string) []string {
	var result []string
	for i := 0; i <= len(s)-n; i++ {
		result = append(result, s[i:i+n])
	}
	return result
}

func UnsafeFirst(n int, s string) string {
	var result string
	for i := 0; i < n; i++ {
		result += string(s[i])
	}
	return result
}

func First(n int, s string) (first string, ok bool) {
	if n > len(s) {
		return "", false
	}
	return UnsafeFirst(n, s), true
}

File: solutions/typescript/series/1/series.ts
export class Series {
  series: string
  constructor(series: string) {
    this.series = series as string
  }

  slices(sliceLength: number): number[][] {
    if (this.series === '') {
      throw new Error("series cannot be empty")
    }
    const seriesLength = this.series.length
    if (sliceLength > seriesLength) {
      throw new Error("slice length cannot be greater than series length")
    }
    if (sliceLength < 0) {
      throw new Error("slice length cannot be negative") 
    }

    if (sliceLength === 0) {
      throw new Error("slice length cannot be zero")
    }
    const result: number[][] = []
    for (let i = 0; i < seriesLength - sliceLength + 1; i++) {
      let slice = this.series.slice(i, i + sliceLength)
      let sliceArr = slice.split('')
      let sliceNum = sliceArr.map(Number)
      result.push(sliceNum)
    }

    return result
  }
}
